packages/asset-swapper/src/utils/aggregate.ts

```diff
@@ -0,0 +1,800 @@
+    EmptyOrders = 'empty orders',
+}
+
+/**
+ * DEX sources to aggregate.
+ */
+export enum ERC20BridgeSource {
+    Native,
+    Uniswap,
+    Eth2Dai,
+    Kyber,
+    NumSources,
+}
+
+/**
+ * Convert a source to a canonical address used by the sampler contract.
+ */
+export const SOURCE_TO_ADDRESS: { [key: string]: string } = {
+    [ERC20BridgeSource.Eth2Dai]: '0x39755357759cE0d7f32dC8dC45414CCa409AE24e',
+    [ERC20BridgeSource.Uniswap]: '0xc0a47dFe034B400B47bDaD5FecDa2621de6c4d95',
+    [ERC20BridgeSource.Kyber]: '0x818E6FECD516Ecc3849DAf6845e3EC868087B755',
+};
+
+/**
+ * Valid sources for market sell.
+ */
+export const SELL_SOURCES = [ERC20BridgeSource.Uniswap, ERC20BridgeSource.Eth2Dai, ERC20BridgeSource.Kyber];
+
+/**
+ * Valid sources for market buy.
+ */
+export const BUY_SOURCES = [ERC20BridgeSource.Uniswap, ERC20BridgeSource.Eth2Dai];
+
+/**
+ * Represents a node on a fill path.
+ */
+export interface Fill {
+    // See `FillFlags`.
+    flags: number;
+    // `FillFlags` that are incompatible with this fill, e.g., to prevent
+    // Kyber from mixing with Uniswap and Eth2Dai and vice versa.
+    exclusionMask: number;
+    // Input fill amount (taker asset amount in a sell, maker asset amount in a buy).
+    input: BigNumber;
+    // Output fill amount (maker asset amount in a sell, taker asset amount in a buy).
+    output: BigNumber;
+    // Fill that must precede this one. This enforces certain fills to be contiguous.
+    parent?: Fill;
+    // Arbitrary data to store in this Fill object. Used to reconstruct orders
+    // from paths.
+    data?: any;
+}
+
+// Internal `data` field for `Fill` objects.
+interface FillData {
+    source: ERC20BridgeSource;
+}
+
+// `FillData` for native fills.
+interface NativeFillData extends FillData {
+    order: SignedOrderWithoutDomain;
+}
+
+/**
+ * Options for `improveMarketBuyAsync()` and `improveMarketSellAsync()`.
+ */
+export interface ImproveOrdersOpts {
+    /**
+     * Liquidity sources to exclude. Default is none.
+     */
+    excludedSources: ERC20BridgeSource[];
+    /**
+     * Whether to prevent mixing Kyber orders with Uniswap and Eth2Dai orders.
+     */
+    noConflicts: boolean;
+    /**
+     * Complexity limit on the search algorithm, i.e., maximum number of
+     * nodes to visit. Default is 1024.
+     */
+    runLimit: number;
+    /**
+     * When generating bridge orders, we use
+     * sampled rate * (1 - bridgeSlippage)
+     * as the rate for calculating maker/taker asset amounts.
+     * This should be a small positive number (e.g., 0.0005) to make up for
+     * small discrepancies between samples and truth.
+     * Default is 0.0005 (5 basis points).
+     */
+    bridgeSlippage: number;
+    /**
+     * Number of samples to take for each DEX quote.
+     */
+    numSamples: number;
+    /**
+     * Dust amount, as a fraction of the fill amount.
+     * Default is 0.01 (100 basis points).
+     */
+    dustThreshold: number;
```

Comment:
nit. prefer `dustAmountThreshhold`

Reply:
idk, we seem to use `amount` for flat amounts, whereas this is a %. Maybe `dustFractionThreshold`?